### PR TITLE
implement the /sock/:sockID endpoint

### LINKED ISSUE #38: implement get sock info
`GET("/:sockId")` should get the socket info corresponding to sockID

### PR DESCRIPTION
**Pull request by jorisSchaller**
This pr implement said endpoint.
Close #38

### CODE REVIEW

File: backend/db/db.go [hunk 1]
```diff
@@ -92,8 +94,39 @@ func editMatchingSock(sockID string, otherSockID string, accept bool) error {
...
+		return Sock{}, fmt.Errorf("corrupted data, unable to read database")
+	}
+
+	//in order to have an empty json array and not a null when converting from the go struct to the json repr
+	// related to TestGetSockInfo@db_test.go
+	//{..
+	//"refusedList": [],
+	//"acceptedList": [],...}
+	if s.AcceptedList == nil {
+		log.Printf("changing type")
```
Comment (Luhrel) [inline backend/db/db.go:121]: Please change to a better log message.

File: backend/db/db.go [hunk 2]
```diff
@@ -92,8 +94,39 @@ func editMatchingSock(sockID string, otherSockID string, accept bool) error {
...
+		s.AcceptedList = make([]string, 0)
+	}
+	if s.RefusedList == nil {
+		log.Printf("changing type")
```
Comment (Luhrel) [inline backend/db/db.go:125]: ditto

File: backend/db/db.go [hunk 3]
```diff
@@ -350,3 +329,36 @@ func DeleteCollection(ctx context.Context, client *firestore.Client,
...
+// 		log.Printf("changing type")
+// 		s.AcceptedList = make([]string, 0)
+// 	}
+// 	if s.RefusedList == nil {
+// 		log.Printf("changing type")
+// 		s.RefusedList = make([]string, 0)
+// 	}
+
+// 	return s, nil
+// }
```
Comment (Luhrel) [inline backend/db/db.go:364]: `git stash` this please

File: backend/db/db_test.go [hunk 4]
```diff
@@ -161,3 +161,29 @@ func TestGetUserSocks(t *testing.T) {
...
+	owner := doc.ID
+
+	s := Sock{
+		ShoeSize:     41,
+		Type:         Profile(1),
+		Color:        "#BEDEAD",
+		Description:  "I tried selling it on onlyFan, but it didn't work",
+		Picture:      "JHAKHSD==",
+		RefusedList:  make([]string, 0), //this init the memory see GetSockInfo@db.go for further detail
+		AcceptedList: make([]string, 0),
```
Comment (Luhrel) [inline backend/db/db_test.go:178]: Please add a test case when `RefusedList` and `AcceptedList` are `nil` ;  they should be properly converted to `make([]string, 0)` (`db.go:121`)

**Review by Luhrel — CHANGES_REQUESTED** [review]

**Review by Luhrel — APPROVED** [review]
